src/tram/tram/ml/base.py

```diff
@@ -228,8 +228,12 @@ def _sentence_tokenize(self, text):
 
     def _extract_pdf_text(self, document):
         with pdfplumber.open(BytesIO(document.docfile.read())) as pdf:
-            text = ''.join(page.extract_text() for page in pdf.pages)
-
+            page_texts = []
+            for page in pdf.pages:
+                page_text = page.extract_text()
+                if page_text:
+                    page_texts.append(page_text)
+            text = ''.join(page_texts)
```

Comment:
I think this can be simplified by appending the guard statement at the end of the for...in loop, like this:
```suggestion
            text = ''.join(page.extract_text() for page in pdf.pages if page.extract_text())
```

Here's an isolated example that shows this works:
```python
class page:
  def __init__(self, text): self.text = text 
  def extract_text(self): return self.text

class pdf:
  pages = [page("hello"),page(None),page("world")]

text = ''.join(page.extract_text() for page in pdf.pages if page.extract_text())
print(text) # helloworld
```